src/utilities/encodedImageStorage/uploadEncodedImage.ts

```diff
@@ -44,6 +40,10 @@ export const uploadEncodedImage = async (
     return encodedImageAlreadyExist.fileName;
   }
 
+  if (previousImagePath) {
+    await deletePreviousImage(previousImagePath);
+  }
+
```

Comment:
Why are we deleting the image if it already exists?
Ideally, we should check here to see if the supplied image differs from the already stored image. If it is, we should delete the old image and add the new one. If not, we can do an early return.

Reply:
> Why are we deleting the image if it already exists?

No, If the supplied image already exists then we are doing an early return with the path of the image.
If the supplied image does not already exists then we are deleting the previous image and updating with the supplied image.

> Ideally, we should check here to see if the supplied image differs from the already stored image. If it is, we should delete the old image and add the new one. If not, we can do an early return.

Yes you are right, and I think this code is doing the same as you mentioned.

```
const encodedImageAlreadyExist = await EncodedImage.findOne({
  content: encodedImageURL,
});

if (encodedImageAlreadyExist) {
  await EncodedImage.findOneAndUpdate(
    {
      content: encodedImageURL,
    },
    {
      $inc: {
        numberOfUses: 1,
      },
    }
  );
  return encodedImageAlreadyExist.fileName;
}
```

Please let me know if I am missing something.